sherpa/astro/ui/utils.py

```diff
@@ -7257,17 +7255,8 @@ def group(self, id=None, bkg_id=None):
 
             # Now check if data is already grouped, and send error message
             # if so
-            if (data.grouped is True):
-                raise DataErr(
-                    'groupset', 'data set', str(self._fix_id(id)), 'True')
-        else:
-            # Just grouping one particular background here
-            if (data.grouped is True):
-                raise DataErr(
-                    'groupset', 'background', str(self._fix_id(bkg_id)), 'True')
-
-        # If we get here, checks showed data not grouped, so set group flag
-        data.group()
+        if not (data.grouped is True):
```

Comment:
I suspect this is intended to catch False or None- if so, pep-8 suggests using   if data.grouped is not True:

Reply:
This is taken from the old code (line 7265, seen in diff above if you expand it a little). I'm not aware of cases where `data.grouped` would be none.